File: clojure/06-io-operations/output.clj
;; Console output in Clojure

;; println adds a trailing newline
(println "Written with println")

;; print does not add a newline
(print "No newline -> ")
(println "same line")

;; prn/pr print values in read-able form (strings keep their quotes)
(prn "quoted")
(prn [1 2 3])

;; println accepts multiple arguments, separated by spaces
(println "Sum of 2 and 3 is" (+ 2 3))

;; format returns a formatted string (Java-style format specifiers)
(println (format "Name: %s, Age: %d" "Ada" 36))
(println (format "Pi is about %.2f" Math/PI))

;; str concatenates its arguments into a single string
(println (str "abc" 1 2 3))

File: php/06-io-operations/output.php
<?php
// echo prints one or more strings as-is
echo "Standard output line\n";

// printf writes a formatted string directly to output
printf("Name: %s, Age: %d\n", "Ada", 36);
printf("Pi is approximately %.2f\n", 3.14159);

// sprintf returns the formatted string instead of printing it
$padded = sprintf("%05d", 42);
echo $padded . "\n";

// Write explicitly to the standard output and standard error streams
fwrite(STDOUT, "Written to STDOUT\n");
fwrite(STDERR, "Written to STDERR\n");
